kedro/config/abstract_config.py

```diff
@@ -19,12 +19,13 @@ def __init__(
         runtime_params: Dict[str, Any] = None,
         **kwargs  # pylint: disable=unused-argument
     ):
+        super().__init__()
         self.conf_source = conf_source
         self.env = env
         self.runtime_params = runtime_params
 
     @abstractmethod  # pragma: no cover
-    def get(self) -> Dict[str, Any]:
+    def get(self) -> Dict[str, Any]:  # type: ignore
```

Comment:
Is this still needed if we are inheriting from `dict` or `UserDict`? I suppose we can completely remove it from here, right?

Reply:
I think the bigger question raised by @noklam and @AntonyMilneQB was if we should actually be inheriting from `dict` or `UserDict` or just keep it to `ABC`: https://github.com/kedro-org/kedro/pull/1870#discussion_r981264452 
https://github.com/kedro-org/kedro/pull/1870#issuecomment-1261131796

I don't have a strong opinion on this, but feel that the intention to make the `ConfigLoader` classes more dict like is clearer if we inherit from `dict`/`UserDict`.

What do you think @idanov ?

Reply:
Agree, let's do it with `UserDict` then.

Reply:
Done!